tests/x11/sshxterm.pm

```diff
@@ -15,16 +15,20 @@ sub run() {
     my $self = shift;
     mouse_hide(1);
     x11_start_program("xterm");
-    script_run("ssh -XC root\@localhost xterm");
-    type_string "yes\n";
-    wait_idle 6;
+
+    # check last incoming ssh traffic, wait until the connection is opened from localhost.
+    # "ESTAB", AND ::1:ssh, AND dport(destination-port) for control with exactitude, that a inc. port from ssh only from localhost is open.
+    # at this moment type password with serenity.
+    script_run("for i in {1..100}; do /usr/sbin/ss -at dport = :ssh | tail -n1 >/dev/$serialdev; sleep 2; done & ");
```

Comment:
try the command with your own pc.
1 time with not your localhost, 2 nd time with your own localhost.
my fix know with precision that when you see the ssh message "type password", the tcp on port 22 get ETABLISHED, so you know that somebody want to connect to ssh, and .*::1 is to ensure that this somebody come from your localhost. i think more stable then this is not possible.